cpp/src/arrow/json/reader-test.cc

```diff
@@ -195,7 +195,7 @@ std::string RowsOfOneColumn(string_view name, std::initializer_list<T> values,
 }
 
 TEST(ReaderTest, MultipleChunksParallel) {
-  int64_t count = 1 << 20;
+  int64_t count = 1 << 10;
```

Comment:
One possible pattern is to choose a different test size if Valgrind is enabled. We do this in other places:
```cpp
#ifdef ARROW_VALGRIND
int64_t count = ...;  // some moderate number
#else
int64_t count = ...;  // some bigger number
#endif
```

Ultimately, it's your choice :-)

Reply:
The test was originally just 1kB. I changed the size around while looking for the benchmark discrepancy, and ended up checking in 1MB. It really doesn't need to be that large, I think